### DISCUSSION TITLE
OMNET++ 6.1.0 + INET 4.5 (in MS Windows at least): Added modules not found unless INET is installed a posteriori

### DISCUSSION THREAD

**Discussion by antoniogrilo** [omnetpp/omnetpp#1256] [General]
I've noticed that when I install INET upon running OMNET++ IDE for the first time, if I add my own modules in src, they are not recognised: upon running an example using those modules, an error window appears saying "Class xxxx not found - perhaps its code was not linked in..."

I have solved the problem by not installing INET 4.5 during first execution of OMNET++, and installing it afterwards, and adding my modules before building it. I don't know if there is another way to solve the problem, but I tried everything: clean, rebuild, etc.

Kind regards,
António Grilo

**Comment by rhornig** [comment]
If you add your own (simple) modules you MUST **rebuild** INET before trying the ne new code. i.e. like with any other code. Before you can try it, you must compile/link it. (use CTRL-B in IDE)

**Reply by antoniogrilo** [reply]
Yes, I know. I did it several times, including clean and then build, changing debug to release, etc., , but didn't work. Only after I have installed INET anew (i.e., not during first OMNET++ run), I was able to have my code integrated in INET.

**Comment by avarga** [comment]
A few tips:
- make sure that your new files show up in the IDE (hit "Refresh" if necessary).
- maybe you has 2 copies of INET installed? you added the code to one, and built/run the other.

**Reply by antoniogrilo** [reply]
Hello,

In fact, the files appeared in the IDE. And I only had/have one copy.

Regards,
António

> A few tips:
> 
> * make sure that your new files show up in the IDE (hit "Refresh" if necessary).
> * maybe you has 2 copies of INET installed? you added the code to one, and built/run the other.
